haystack/nodes/connector/crawler.py

```diff
@@ -56,6 +58,7 @@ def __init__(
         extract_hidden_text=True,
         loading_wait_time: Optional[int] = None,
         crawler_naming_function: Optional[Callable[[str, str], str]] = None,
+        webdriver_options: Optional[List[str]] = None,
```

Comment:
I wonder if we should make this a `Tuple[str] = ("--headless", "--disable-gpu", "--disable-dev-shm-usage", "--single-process")` to make it clear from the function signature there are set defaults?

Reply:
We could then convert this to a list `webdriver_options = list(webdriver_options)` in the function before appending any additional options.

Reply:
I've got warnings on Sonar regarding initializing the parameter on the function signature (S5717), as it will be altered inside the function.

> a default value is either modified in the function or assigned to anything else

Don't you think we could follow its guidelines? Initializing with None and then setting inside function? As the default parameters is of low interest to the user, if he has any specific scenario, he will set the parameter anyway.

Reply:
Maybe we could change the naming in documentation to forced parameters or initial.

Reply:
Yeah good point let's stick with the python guidelines here and use `None` as the default.